sw/simulator/nps/nps_fdm_gazebo.cpp

```diff
@@ -329,6 +331,15 @@ static void gazebo_read(void)
   /* Fill FDM struct */
   fdm.time = world->GetSimTime().Double();
 
+  // Find world acceleration by differentiating velocity
+  // model->GetWorldLinearAccel() does not seem to take the velocity_decay into account!
+  // Derivation of the velocity also follows the IMU implementation of Gazebo itself:
+  // https://bitbucket.org/osrf/gazebo/src/e26144434b932b4b6a760ddaa19cfcf9f1734748/gazebo/sensors/ImuSensor.cc?at=default&fileviewer=file-view-default#ImuSensor.cc-370
+  float dt = fdm.time - time_prev;
```

Comment:
Shouldn't this be a double?  fdm.time and time_prev are both doubles. gazebo::math::Vector3 also contain doubles.

Reply:
Yes, fixed